chapter12/dags/nyc_dag.py

```diff
@@ -104,10 +104,10 @@ def transform_citi_bike_data(df):
             crs="EPSG:4326",
             geometry=geopandas.points_from_xy(df["end_station_longitude"], df["end_station_latitude"]),
         )
-        df_with_zones = geopandas.sjoin(start_gdf, taxi_zones, how="left", op="within").rename(
+        df_with_zones = geopandas.sjoin(start_gdf, taxi_zones, how="left").rename(
             columns={"LocationID": "start_location_id"}
         )
-        end_zones = geopandas.sjoin(end_gdf, taxi_zones, how="left", op="within")
+        end_zones = geopandas.sjoin(end_gdf, taxi_zones, how="left")
         df_with_zones["end_location_id"] = end_zones["LocationID"]
```

Comment:
In general the use of Pandas is a bit updated throughout this chapter.
They are all valid and will run,  but could be way better.

For example direct assignments like `df_with_zones["end_location_id"] = end_zones["LocationID"]` should be done with the .assign() WOW. 

For example, the function `transform_taxi_data` could be greatly improved with a good refactoring.

I can definitively help out, lets discuss it about it today in the meeting